### ISSUE TITLE
One More Case where 'async_recursion is required

### ISSUE THREAD

**Issue by Sythanis** [dcchut/async-recursion#7]
Hey again @dcchut . Just found another case when a manual overwrite of the lifetime is required. I am having a function simillar to 

```rust
pub async fn execute<T>(&self, executor: &'async_recursion E)  
where
  E: Executor + Send + Sync
{
  *self.counter.lock().await -=1;

  if self.counter.lock().await > 0 {
    self.execute(executor).await;
  }
}
```

Notice that I had to stipulate `'async_recursion` in it. If I remove it I get a wierd error, saying that `'life0` is doubly defined. Have you came accross this case before? 
```text
error[E0263]: lifetime name `'life0` declared twice in the same scope
```

**Comment by Sythanis** [comment]
I tried using `cargo-expand`. I get something like that
```
pub fn execute<'life0, 'life0, 'async_recursion, E>(
        &'life0 self,
        executor: &'life0 E,
...
```

That seems regiression from #6 . I confirmed that before the last push this was not there

**Comment by dcchut** [comment]
Thanks for the report @Sythanis - looks like I didn't have any test coverage for this case.  I've released v0.3.1 which fixes this issue.
